src/planner/rules/expr.rs

```diff
@@ -85,6 +85,22 @@ pub fn rules() -> Vec<Rewrite> { vec![
     rw!("if-not";    "(if (not ?cond) ?then ?else)" => "(if ?cond ?else ?then)"),
 
     rw!("avg";       "(avg ?a)" => "(/ (sum ?a) (count ?a))"),
+
+    // Extract Common Predicate
+    // example:
+    //            OR
+    //          /    \
+    //      AND         AND
+    //     /   \       /  \
+    // a = b   c = 1  a = b  d = 2
+    //
+    // After rule:
+    //             AND
+    //            /   \
+    //         a = b   OR
+    //                /  \
+    //            c = 1  d = 2
+    rw!("extract-common-predicate"; "(or (and ?a ?b) (and ?a ?c)))" => "(and ?a (or ?b ?c))"),
```

Comment:
"extract-common-predicate" This name looks too broad to me.
I asked ChatGPT and it said this is "Distributive Law of Disjunction".
Do you think it's a better name?

```suggestion
    rw!("add-or-distri"; "(or (and ?a ?b) (and ?a ?c)))" => "(and ?a (or ?b ?c))"),
```